gradio/cli/commands/components/_docs_utils.py

```diff
@@ -0,0 +1,892 @@
+from __future__ import annotations
+
+import inspect
+import json
+import re
+import types
+import typing
+from subprocess import PIPE, Popen
+
+
+def find_first_non_return_key(some_dict):
+    """Finds the first key in a dictionary that is not "return"."""
+    for key, value in some_dict.items():
+        if key != "return":
+            return value
+    return None
+
+
+def format(code: str, type: str):
+    """Formats code using ruff."""
+    if type == "value":
+        code = f"value = {code}"
+
+    ruff_args = ["ruff", "format", "-", "--line-length=60"]
+
+    process = Popen(
+        ruff_args,
+        stdin=PIPE,
+        stdout=PIPE,
+        stderr=PIPE,
+        universal_newlines=True,
+    )
+    formatted_code, err = process.communicate(input=code)
+    print(err)
```

Comment:
I'm getting that `--line-length` is not a valid ruff parameter name

error: unexpected argument '--line-length' found

<img width="614" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/a2633e14-f9c0-41ce-94b7-f671ad107b61">

Just FYI

Reply:
How are you switching your python version. `Line-length` _is` a valid parameter. I'm guessing that your ruff version for that python version is earlier than 0.1.4 when the param was added or maybe even prior to 0.1.2 when the formatter was added.

I've added a recent version of ruff as a dep now, so this shouldn't be an issue.